File: backend/app/routers/content.py
from fastapi import APIRouter, HTTPException
from pathlib import Path
import json

router = APIRouter(prefix='/content', tags=['content'])

DATA_DIR = Path(__file__).parent.parent / 'data'


@router.get('/lessons')
def get_lessons():
    data = json.loads((DATA_DIR / 'lessons.json').read_text(encoding='utf-8'))
    return data


@router.get('/lessons/{lesson_id}')
def get_lesson(lesson_id: str):
    data = json.loads((DATA_DIR / 'lessons.json').read_text(encoding='utf-8'))
    lesson = next((l for l in data['lessons'] if l['id'] == lesson_id), None)
    if not lesson:
        raise HTTPException(status_code=404, detail='Lesson not found')
    return lesson


@router.get('/gallery')
def get_gallery():
    data = json.loads((DATA_DIR / 'gallery.json').read_text(encoding='utf-8'))
    return data

File: backend/src/routers/content.ts
import { Router } from 'express'
import { readFileSync } from 'fs'
import { fileURLToPath } from 'url'
import { dirname, join } from 'path'

const __dirname = dirname(fileURLToPath(import.meta.url))
const DATA_DIR = join(__dirname, '..', 'data')

const router = Router()

router.get('/lessons', (_req, res) => {
  const data = JSON.parse(readFileSync(join(DATA_DIR, 'lessons.json'), 'utf-8'))
  res.json(data)
})

router.get('/lessons/:id', (req, res) => {
  const data = JSON.parse(readFileSync(join(DATA_DIR, 'lessons.json'), 'utf-8'))
  const lesson = data.lessons.find((l: { id: string }) => l.id === req.params.id)
  if (!lesson) return res.status(404).json({ detail: 'Lesson not found' })
  res.json(lesson)
})

router.get('/gallery', (_req, res) => {
  const data = JSON.parse(readFileSync(join(DATA_DIR, 'gallery.json'), 'utf-8'))
  res.json(data)
})

export default router
